feat: addmovie api 연동

depoo69 opened the pull request:
**📌 관련 이슈번호**
#58 
---

### 체크리스트

- [x]  🎋 base 브랜치를 dev 브랜치로 설정했나요?
- [x]  🖌️ PR 제목은 형식에 맞게 잘 작성했나요?
- [x]  🏗️ 빌드는 성공했나요? (yarn build)
- [x]  🧹 불필요한 코드는 제거했나요? e.g. console.log
- [x]  🙇‍♂️ 리뷰어를 지정했나요?
- [x]  🏷️ 라벨은 등록했나요?

---

## ✅ Key Changes

> 이번 PR에서 작업한 내용을 간략히 설명해주세요
> MovieManagement 페이지에 AdminEditModal 추가 (api연동 필요)
   영화 이미지 및 정보를 수정할 수 있는 기능 구현

---

**📢 To Reviewers**

---

**📸 스크린샷 or 실행영상**

depoo69 committed:
feat: addmovie api 연동

depoo69 committed:
feat: addmovie api 연동 및 모달 추가

gwagjiug commented on src/apis/admin/addmovie/addMovie.js:
```diff
@@ -0,0 +1,59 @@
+import { postRefresh } from "../../users/user";
```
일단 폴더명은 도메인별로 나누는 게 좋을듯 addMovie 보다 admin 폴더안에 addMovie가 있는게 더 자연스러움

depoo69 replied:
addmovie폴더를 삭제하고 admin폴더에 src/apis/admin/addmovie.js, queries.js, searchMovies.js 이런식으로 해놓음
앞으로 관리자 페이지 API 파일들 이쪽에 놓으면 될거같애

gwagjiug commented on src/apis/admin/addmovie/addMovie.js:
```diff
@@ -0,0 +1,59 @@
+  }
+
+  // 요청 데이터 준비
+  const payload = selectedMovies.map((movie) => ({
```
이 코드는 왜 필요한건지??

depoo69 replied:
서버에 보낼 데이터를 생성하는 과정인데, selectedMovies가 영화 데이터 원본 배열이고,
이걸 서버가 요구하는 형식에 맞게 데이터를 가공해야됨. 그래서 사실 검색 결과에 필요한 데이터인 
영화이름(title), 평균평점(일단 rating), 장르(genre), 개봉일(repRlsDate) 정도만 필요하기 때문에 
나머지는 다 지워도 되는데, 근데 생각해보니깐 영화를 등록해서 우리 사이트에서 사용할 거면 
출연진(staff), 줄거리(plots) 이런 것들이 영화 상세에서도 쓰이고 그래서 안건들고 일단 놔두려고 함

gwagjiug commented on src/apis/admin/addmovie/queries.js:
```diff
@@ -0,0 +1,23 @@
+
+export const useSearchMovies = (searchTerm) => {
+  return useQuery({
+    queryKey: ['searchMovies'],
```
queryKey 도 searchMovies 보다는 `admin` 이 포함된 이름이면 더 좋을 것 같음 항상 도메인이 무엇인지 생각해보셈 

adminSearch 가 더 나을려나

depoo69 replied:
useAdminSearchMovies 로 변경완료

gwagjiug commented on src/apis/admin/addmovie/queries.js:
```diff
@@ -0,0 +1,23 @@
+
+export const useRegisterMovies = () => {
+  return useMutation({
+    mutationKey: ['registerMovies'],
```
얘도 마찬가지

depoo69 replied:
useAdminRegisterMovies 로 변경완료

gwagjiug commented on src/apis/admin/addmovie/searchMovies.js:
```diff
@@ -0,0 +1,44 @@
+import { postRefresh } from "../../users/user";
+
+export const fetchMovies = async (searchTerm) => {
+  let accessToken = localStorage.getItem('accessToken');
```
사실 이 코드도 만약 같은 폴더 안에 있었다면 accessToken도 두번 가져올 필요가 없겠죠? 근데 코드가 너무 길어서 뭐.. 이것도 나쁘지않은 것 같은데 한번 고민해보셈

depoo69 replied:
고민해보겠습니당

gwagjiug commented on src/components/common/admin/AddMovie.jsx:
```diff
@@ -0,0 +1,268 @@
+
+  // 서버에서 받은 영화 데이터를 안전하게 처리
+  const movies =
+    data?.data.map((movie) => ({
```
이 코드는 왜 필요한거? 코드만봐서 잘 모르겠음

depoo69 replied:
서버에서 가져온 데이터를 클라이언트에서 사용할 수 있게 변환하는 건데,
 Main.jsx의 const UpcomingData = result?.data?.content || []; 이것처럼 
data?.data 이걸로 필요한 정보만 추려내서 컴포넌트에서 사용하게끔 하는 역할

gwagjiug commented on src/components/common/admin/AddMovie.jsx:
```diff
@@ -0,0 +1,268 @@
+
+  const handleRegister = () => {
+    const selectedMovieData = movies
+      .filter((movie) => selectedMovies.includes(movie.id))
```
이것도 마찬가지 무슨코드인지 설명 댓글로 달아주셈

gwagjiug commented on src/components/common/admin/AddMovie.jsx:
```diff
@@ -0,0 +1,268 @@
+    });
+  };
+
+  const handlePageChange = (event, value) => {
```
전체적으로 이 부분에 코드와 함수 선언이 너무 많음 util 폴더에 함수를 따로 선언해서 가독성도 재사용성도 높일 수 있음 나도 잘하진 않지만 Register.jsx 파일을 한번 찾아보셈

gwagjiug commented on src/components/common/admin/AddMovie.jsx:
```diff
@@ -0,0 +1,268 @@
+        onClick={handleRegister}
+        disabled={selectedMovies.length === 0 || isRegistering}
+      >
+        {isRegistering ? '등록 중...' : '등록'}
```
쿼리는 자동으로 isLoading 을 제공해주기 때문에 query에서 제공하는 isLoading 을 한번 사용해보면 좋을듯

gwagjiug commented on src/components/common/admin/AddMovie.jsx:
```diff
@@ -0,0 +1,268 @@
+  }),
+
+  ClearButton: styled(Button)({
+    color: lightTheme.color.buttonPink,
```
lightTheme 만 사용할 수 있음?? 기현이형꺼보면 MUI에도 ThemeProvider 사용했던데 한번 시도해보셈 되게 비효율적인 방법이긴함 고쳐보셈

gwagjiug commented on src/components/common/admin/LikeManagement.jsx:
```diff
@@ -0,0 +1,199 @@
+      genre: rest.genre || 'N/A',
+      createdDate: rest.createdDate || '2024-12-01',
+    }));
+    setMovies(enhancedMovies);
```
여기도 윗페이지랑 마찬가지임 query를 쓰면 useEffect를 사용하지 않아도 데이터패칭이 가능함

gwagjiug commented on src/components/common/admin/MovieManagement.jsx:
```diff
@@ -0,0 +1,246 @@
+    updateMovieField(movieId, field, value);
+  };
+
+  const handleSearch = (term) => {
```
여기도 코드 너무 긺

gwagjiug reviewed (changes requested):
리뷰하다가 포기함 코드가 너무 긴것도 있고 MUI에 lightTheme 말고 ThemeProvider 를 사용해보셈 기현이형은 했던데 왜 안되는지 한번 생각해보고 되게만들려고 한번 노력해보셈

진짜 실력 늘을 거임 중간에 코드리뷰 하다가 그냥 말았는데 모든 페이지가 거의 비슷함 함수 분리가 아예 안되어있고 의도를 알 수 없는 코드도 엄청 많음 내가 댓글 단거에 대해서 답변 부탁 (무슨 코드인지)

depoo69 committed:
fix: addmovie API 폴더 구조 변경 및 AddMovie 스타일 변경

depoo69 committed:
stlye: MovieManagement 입력 필드 삭제 및 UI 변경

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
fix: 관리자페이지 router 설정

depoo69 committed:
fix: AddMovie 검색 및 추가 기능 수정

depoo69 committed:
feat: 리뷰관리 API 연동

depoo69 committed:
feat: 리뷰 블라인드 기능 API 연동

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
fix: 영화 관리 검색 기능 수정

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
feat: 영화 삭제 API 연동, 영화 정보 수정 API 연동,
포스터 이미지 업로드 S3 API 연동,영화별 좋아요 수 수정 API 연동
fix: 영화 추가에서 영화 추가 시 개봉일 정보가 안들어오던 문제 수정, 리뷰 관리에서
특정 조건의 리뷰만 보여주던 문제 수정,
style: 리뷰 관리에서 삭제버튼 제거, 영화 추가에서 평점 필드 관람 등급으로 수정, 영화추가에서
개봉일 미정 시 필드를 비워놓도록 수정

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
refactor: 함수 분리

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

depoo69 committed:
style: 관리자 랭크 조작 컴포넌트 추가 및 API 연동

depoo69 committed:
fix: 사용되지 않는 AdminLoginPage 삭제

depoo69 committed:
fix: 안쓰이는 파일 CommentBadge 삭제

depoo69 committed:
fix: RankTransferList.js 함수 부분 utils파일로 변경

depoo69 committed:
style: 랭크 관리 transferlist 수정

depoo69 committed:
Merge branch 'dev' of https://github.com/LG-URECA-FINAL-TEAM8/filmeet-frontend into FM-58

gwagjiug committed:
fix : query key 수정

gwagjiug committed:
fix : page size 수정

gwagjiug committed:
feat : admin 로그인시 navigate
